src/Domain/CoordinateMaps/UniformCylindricalSide.cpp

```diff
@@ -0,0 +1,752 @@
+  double function_at_lambda_max = function_to_zero(lambda_max);
+
+  if (1.0 - lambda_min <
+             std::numeric_limits<double>::epsilon() * 100.0) {
+    // lambda_min is within roundoff of lambda_max, meaning that the
+    // point must be within roundoff of a point on the outer sphere
+    // where there is only one possible value of lambda.  Here we
+    // treat this case as if lambda is exactly 1.0.
+    lambda = 1.0;
+  } else {
+    // Here we do a root find.
+    if (function_at_lambda_min * function_at_lambda_max > 0.0) {
+      // Root not bracketed.
+      // If the bracketing failure is due to roundoff, then
+      // slightly adjust the bounds to increase the range. Otherwise, error.
+      //
+      // roundoff_ratio is the limiting ratio of the two function values
+      // for which we should attempt to adjust the bounds.  It is ok for
+      // roundoff_ratio to be much larger than actual roundoff, since it
+      // doesn't hurt to expand the interval and try again when
+      // otherwise we would just error.
+      constexpr double roundoff_ratio = 1.e-3;
+      if (abs(function_at_lambda_min) / abs(function_at_lambda_max) <
+          roundoff_ratio) {
+        // Slightly decrease lambda_min.  How far do we decrease it?
+        // Figure that out by looking at the deriv of the function.
+        const double deriv_function_at_lambda_min =
+            function_and_deriv_to_zero(lambda_min).second;
+        const double trial_lambda_min_increment =
+            -2.0 * function_at_lambda_min / deriv_function_at_lambda_min;
+        // new_lambda_min = lambda_min + trial_lambda_min_increment would be
+        // a Newton-Raphson step except for the factor of 2 in
+        // trial_lambda_min_increment. The factor of 2 is there to
+        // over-compensate so that the new lambda_min brackets the
+        // root.
+        //
+        // But sometimes the factor of 2 is not enough when
+        // trial_lambda_min_increment is roundoff-small so that the
+        // change in new_lambda_min is zero or only in the last one or
+        // two bits.  If this is the case, then we set
+        // lambda_min_increment to some small roundoff value with the
+        // same sign as trial_lambda_min_increment.
+        // Note that lambda is always between zero and one, so it is
+        // ok to use an absolute epsilon.
+        const double lambda_min_increment =
+            abs(trial_lambda_min_increment) >
+                    1000.0 * std::numeric_limits<double>::epsilon()
+                ? trial_lambda_min_increment
+                : std::copysign(1000.0 * std::numeric_limits<double>::epsilon(),
+                                trial_lambda_min_increment);
+        const double new_lambda_min = lambda_min + lambda_min_increment;
+        const double function_at_new_lambda_min =
+            function_to_zero(new_lambda_min);
+        if (function_at_new_lambda_min * function_at_lambda_min > 0.0) {
+          ERROR(
```

Comment:
Apparently need an error test to test this situation? Is that hard to do?

Reply:
I told codecov to ignore this error test and the others too.  Those are all cases that should never happen unless there is a bug, so I have no idea how to cause them to happen.